applications/vanilla/views/discussion/helper_functions.php

```diff
@@ -481,7 +481,11 @@ function WriteEmbedCommentForm() {
          );
          echo $Controller->Form->Button('Post Comment', array('class' => 'Button CommentButton'));
       } else {
-         $AuthenticationUrl = SignInUrl($ReturnUrl);
+         $root = '';
+         if (val('vanilla_forum_url', $Controller->Data('ForeignSource'))) {
+            $root = trim(val('vanilla_forum_url', $Controller->Data('ForeignSource'), ''), '/');
+         }
+         $AuthenticationUrl = $root.SignInUrl($ReturnUrl);
```

Comment:
If you want to add the current domain then you want to use `Url($AuthenticationUrl, true)` and rollback the other file changes you made.